src/System.Management.Automation/security/CatalogHelper.cs

```diff
@@ -477,90 +468,92 @@ internal static string CalculateFileHash(string filePath, string hashAlgorithm)
         /// <returns>Dictionary mapping files relative paths to HashValues.</returns>
         internal static Dictionary<string, string> GetHashesFromCatalog(string catalogFilePath, WildcardPattern[] excludedPatterns, out int catalogVersion)
         {
-            IntPtr resultCatalog = NativeMethods.CryptCATOpen(catalogFilePath, 0, IntPtr.Zero, 1, 0);
-            IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
             Dictionary<string, string> catalogHashes = new Dictionary<string, string>(StringComparer.CurrentCultureIgnoreCase);
             catalogVersion = 0;
 
-            if (resultCatalog != INVALID_HANDLE_VALUE)
+            SafeCATHandle resultCatalog;
+            try
             {
-                try
+                resultCatalog = WinTrustMethods.CryptCATOpen(catalogFilePath, 0, IntPtr.Zero, 1, 0);
+            }
+            catch (Win32Exception e)
+            {
+                ErrorRecord errorRecord = new ErrorRecord(
+                    new InvalidOperationException(StringUtil.Format(CatalogStrings.UnableToOpenCatalogFile, catalogFilePath), e),
+                    "UnableToOpenCatalogFile",
+                    ErrorCategory.InvalidOperation,
+                    null);
+                _cmdlet.ThrowTerminatingError(errorRecord);
+                return catalogHashes;
+            }
+
+            using (resultCatalog)
```

Comment:
The same. No need to use the `using`.

Reply:
Same thing, just explicitly ensures it gets disposed when no longer needed instead of relying on GC.